src/ORM/Association/HasMany.php

```diff
@@ -233,10 +233,8 @@ protected function _saveTarget(
             }
 
             if (!empty($options['atomic'])) {
-                if ($original[$k] instanceof EntityInterface && $entity instanceof EntityInterface) {
```

Comment:
This check was redundant as `$entity instanceof EntityInterface` is the 1st thing that's checked as start of this loop.